### ISSUE TITLE
Vega - Advanced Cross-Filtering

### ISSUE THREAD

**Issue by tomatminceddata** [deneb-viz/deneb#504]
**Aim**: I want the current selected "Year Month" to be used as a cross-filter in combination with the selected country!
**Disclaimer**: I think I want to achieve something advanced, but I assume that my current Deneb/Vega skills are not on par with what I want, for this reason, if I discover an issue or do something stupid.

The "Year Month" will be changed by a Timer event inside Vega.

Basically everything works (including CTRL-key to select multiple countries).. But when initializing/calling the cross-filtering by selecting the the first country, the current timer event will be reset.

To start/stop the timer event these steps are required:
1. Move to the first "Year Month", this is achieved by "double clicking inside the view.
2. start the timer by alt+doubleclick
3. stop the timer by alt+doubleclick

When I want to pass the current "Year month" to pbiCrossFilterApply(...) I stop the timer (alt+doubleclicking) and "shift+ctrl+alt+click" one of the countries.

The current "Year Month" is passed to pbiCrossFilterApply(...), but
at the same moment the counter will be reset to the latest "Year Month:" - "April 2024"

If I then select a 2nd country the 2nd country and "April 2024" will be added to the cross-filter.

I want to get rid of "resetting the counter when selecting a counry"

There is another event (wanted)

"shift+doublecliick" sets the counter to the last element "April 2024"

I added a gif showcasing what I'm talking about ".
![Deneb - Vega - Advanced Cross Filtering](https://github.com/user-attachments/assets/4548beb6-0b3f-437a-bd3c-91d2e5d4919a)
"

Here is the pbix: https://tommartens-my.sharepoint.com/:u:/g/personal/tom_minceddata_com/EbSwD0kVvq9LiSUkwbBmneoBeI6r6wGzCM8uowkzdvtGzw?e=bNdIIL

**Edit on 2024-09-28**
The pbix file now contains a new page "" with a simplified Deneb/Vega visualization. I think this visual is better suited to understand what I'm talking about, what I'm doing wrong, or maybe helping to fix an issue.
What I want: When passing the cross-filtering to the tiny Power BI table, I want the rectangle to stay red.

**Comment by dm-p** [comment]
If you don't want signals to be reset on canvas updates, this will be dependent on #410

It is a very significant change to achieve this but 2.0's architecture will be based on preserving the Vega view for the lifetime of the visual (or from when a specification change is applied) and this will provide you with the functionality you want.

I'll close this issue as it's effectively a duplicate of #410 - you can keep an eye on that issue or add requirements to that one so I can consider what's needed from a user perspective

**Comment by Giammaria** [comment]
Ah, yes! That’s the issue that I was looking for. I can appreciate the LOE required for this. Once implemented, even if not for a very long time, it will allow us to do some pretty significant joining and cross-filtering.Thanks.MadisonSent from my iPhoneOn Sep 29, 2024, at 3:08 AM, Daniel Marsh-Patrick ***@***.***> wrote:﻿
If you don't want signals to be reset on canvas updates, this will be dependent on #410
It is a very significant change to achieve this but 2.0's architecture will be based on preserving the Vega view for the lifetime of the visual (or from when a specification change is applied) and this will provide you with the functionality you want.
I'll close this issue as it's effectively a duplicate of #410 - you can keep an eye on that issue or add requirements to that one so I can consider what's needed from a user perspective
